src/panoramic/cli/transform/executor.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+from dbt.exceptions import FailedToConnectException
+
+from panoramic.cli.connections import (
+    Connections,
+    fill_dbt_required_connection_keys,
+    get_dialect_credentials,
+)
+from panoramic.cli.errors import ConnectionNotFound, TransformExecutionFailed
+from panoramic.cli.print import echo_info
+from panoramic.cli.transform.pano_transform import CompiledTransform
+
+logger = logging.getLogger(__name__)
+
+
+class TransformExecutor:
+    @classmethod
+    def execute(cls, compiled_transform: CompiledTransform):
+        connection_name = compiled_transform.transform.connection_name
+        try:
+            connection_credentials_dict = Connections.get_by_name(connection_name)
+        except ValueError:
+            raise ConnectionNotFound(connection_name)
+
+        connection_credentials_dict = fill_dbt_required_connection_keys(connection_credentials_dict)
+        credentials, credential_error = get_dialect_credentials(connection_credentials_dict)
+
+        if credential_error is not None:
+            # FIXME: raise a better error
+            echo_info(f'{connection_name} FAIL: {credential_error}...')
+            return
+
+        try:
+            logger.debug(f'Executing transform {compiled_transform.transform.name} on {connection_name}')
+            Connections.execute(sql=compiled_transform.compiled_query, credentials=credentials)
```

Comment:
Would it make sense to have a class for executing SQL that handles/pools connections? Right now each time we call Connections.execute a new connection is created and SQL is executed. IMHO ideally we'd separate connection handling from actual SQL execution and from transform logic